sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -322,6 +322,15 @@ object SQLConf {
       .booleanConf
       .createWithDefault(true)
 
+  // We can add LZO after Apache ORC 1.4.0 is used.
+  val ORC_COMPRESSION = buildConf("spark.sql.orc.compression.codec")
+    .doc("Sets the compression codec use when writing ORC files. Acceptable values include: " +
+      "uncompressed, snappy, zlib.")
+    .stringConf
+    .transform(_.toLowerCase(Locale.ROOT))
+    .checkValues(Set("uncompressed", "snappy", "zlib"))
```

Comment:
Hm, but `lzo` still seeks `org.apache.hadoop.hive.ql.io.orc.LzoCodec` though. Wouldn't it be better to leave as what it is intended for now if I understood correctly? I think it should be almost seldom but disallowing `lzo` actually looks disallowing the chance of a custom use allowed via `.option()` API.

I think `none` was introduced as a compression key for `.option()` first for Parquet and it was matched for consistency. I think we have kept `none` for `.option()` API specific somehow.

Reply:
Yep. "none" is added back. "lzo" is not supported. It failed in the current master branch. We have a ignored test case for that LZO failure.